src/main/java/org/gridsuite/sensitivityanalysis/server/service/nonevacuatedenergy/NonEvacuatedEnergyInputBuilderService.java

```diff
@@ -0,0 +1,514 @@
+            // get branch from network
+            Branch branch = network.getBranch(monitoredEquipment.getId());
+            if (branch == null) {  // branch not found : just report and ignore the branch
+                reporter.report(Report.builder()
+                    .withKey("monitoredBranchNotFound")
+                    .withDefaultMessage("Could not find the monitored branch ${id}")
+                    .withSeverity(TypedValue.ERROR_SEVERITY)
+                    .withValue("id", monitoredEquipment.getId())
+                    .build());
+                return;
+            }
+            Optional<CurrentLimits> currentLimits1 = branch.getCurrentLimits1();
+            Optional<CurrentLimits> currentLimits2 = branch.getCurrentLimits2();
+            if (currentLimits1.isEmpty() && currentLimits2.isEmpty()) {
+                // no current limits on both sides found : report and throw exception
+                reporter.report(Report.builder()
+                    .withKey("monitoredBranchNoCurrentLimits")
+                    .withDefaultMessage("No current limits for the monitored branch ${id}")
+                    .withSeverity(TypedValue.ERROR_SEVERITY)
+                    .withValue("id", monitoredEquipment.getId())
+                    .build());
+                throw new PowsyblException("Branch '" + branch.getId() + "' has no current limits !!");
+            }
+
+            MonitoredBranchThreshold monitoredBranchThreshold = new MonitoredBranchThreshold(branch);
+
+            if (branches.isIstN() || branches.isIstNm1()) {  // Ist activated : we consider the permanent limit
+                // permanent limit on side 1
+                boolean factors1Generated = genFactorForPermanentLimit(branch, Branch.Side.ONE, currentLimits1, variableIds, variableSet, sensitivityVariableType, branches, monitoredBranchThreshold, contingencies, result, reporter);
+
+                // permanent limit on side 2
+                boolean factors2Generated = genFactorForPermanentLimit(branch, Branch.Side.TWO, currentLimits2, variableIds, variableSet, sensitivityVariableType, branches, monitoredBranchThreshold, contingencies, result, reporter);
+
+                if (!factors1Generated && !factors2Generated) {
+                    // no temporary limit on one side : report and throw exception
+                    reporter.report(Report.builder()
+                        .withKey("monitoredBranchNoPermanentLimits")
+                        .withDefaultMessage("No permanent limits for the monitored branch ${id}")
+                        .withSeverity(TypedValue.ERROR_SEVERITY)
+                        .withValue("id", monitoredEquipment.getId())
+                        .build());
+                    throw new PowsyblException("Branch '" + branch.getId() + "' has no permanent limits !!");
+                }
+            }
+
+            if (StringUtils.isNotEmpty(branches.getLimitNameN())) {  // here we consider the temporary limits
```

Comment:
this does not seem to work, if I enable the N IST and give a N limit, then the N limit is taken into while it should not.
maybe this should be 
```if (branches.isIstN())``` ?